### DISCUSSION TITLE
Reduce repo size

### DISCUSSION THREAD

**Discussion by CodeBleu** [cncf/k8s-conformance#2817] [General]
Is there a better way to do things?  Cloning this repo results ina `7.2G` directory!!  That is even with doing a `git clone --depth 1....`

**Comment by riaankleinhans** [comment]
Hi @CodeBleu thank for pointing this out!
Very good point.

We are going to implement a few change to make this happen.
- We are going to limit the files that could be submitted in a PR to only the 4 files requiered for comformance verification. That would slow the growth of the repo
- We are going to archive all submissions older than 2 year
- We would also looking are a process that would continually archive older submissions as time pass.

My aim is to get this done before the end of this release.

**Comment by riaankleinhans** [comment]
Step 1 is done. The CNCF-CI bot now only allow 4 files requiered to varify conforamcne. This will reduce the future growth of the repo size.
We will address achieving next.
![image](https://github.com/cncf/k8s-conformance/assets/61125752/c216b676-0692-4d95-8312-ff1333cd5e7e)

**Comment by riaankleinhans** [comment]
The repo have been cleaned up and file <1.23 moved to an archive repo.
The repo is now less and 3GB to clone.
@CodeBleu I trust we can close this discussion?

**Reply by CodeBleu** [reply]
Where is the archive repo you speak of, and where in this repo is talking about those changes?

I still think that multiple GB for a git repo is not good.  I guess the only way this is going to get smaller is a periodic run of archiving things that are older than 2 years?  Is this something this is happening? If so, then I would hope that however often this is ran it should be making the repo smaller each time?

**Reply by riaankleinhans** [reply]
The referance will be added here: https://github.com/cncf/k8s-conformance/pull/2867
And yes, the plan is to keep round 2 years worth of information before moving it out to the archive repo.

**Reply by CodeBleu** [reply]
Just curious how often the purge of > 2 years happens? Once a month?

**Reply by riaankleinhans** [reply]
After every release cycle, when a new release come in like 1.29 end of the year, 1.23 will be dropped from the main repo and April 2024 when 1.30 release we drop 1.24 etc.
The aim would be to add automation that run but for the moment it would be done manually.

**Comment by BobyMCbobs** [comment]
Since this, I've reduced the repo size!
It's now 2.7GB, which is 5.1GB less.
